src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```
@@ -559,6 +561,60 @@ private static String convertUpperCamelToUpperUnderscore(String text) {
         return result.toString();
     }
 
+    public ParseStatus parseJavadocComment(DetailAST javadocCommentAst) {
+        blockCommentLineNumber = javadocCommentAst.getLineNo();
+
+        final String javadocComment = JavadocUtil.getJavadocCommentContent(javadocCommentAst);
+
+        final ParseStatus result = new ParseStatus();
+
+        try {
+            final JavadocCommentsParser javadocParser = createJavadocParser(javadocComment);
+
+            final JavadocCommentsParser.JavadocContext javadoc = javadocParser.javadoc();
+
+            final DetailNode tree = new JavadocCommentsAstVisitor().visit(javadoc);
+
+            // adjust first line to indent of /**
+            adjustFirstLineToJavadocIndent(tree,
+                        javadocCommentAst.getColumnNo()
+                                + JAVADOC_START.length());
+            result.setTree(tree);
+        }
+        catch (ParseCancellationException | IllegalArgumentException exc) {
+        }
```

Hiding exceptions isn't a good idea. It brings to question how do we know for sure your code is working correctly.

---

Would it make more sense for now to just rethrow the exception as an Error so we can at least surface it clearly during development? Once we finalize the grammar, we can handle the exceptions and parsing errors in this catch block.

---

> just rethrow the exception as an Error so we can at least surface it clearly during development

I am good with that. I don't really understand why we don't use similar code from the original grammar or the java grammar.

---

> I don't really understand why we don't use similar code from the original grammar or the java grammar.

The original grammar includes logic in this catch block that handles specific error scenarios. for example, detecting missing HTML tags and generating detailed ParseErrorMessage instances. This level of error handling hasnt been implemented yet in the new grammar, which is why we are not using similar code here now.

Also, the overall error handling logic might change as the new grammar evolves, so we are planning to revisit this in a dedicated PR toward the end of the grammar update.

 code of the old grammar:

 ```Java
catch (ParseCancellationException | IllegalArgumentException exc) {
            ParseErrorMessage parseErrorMessage = null;

            if (exc.getCause() instanceof FailedPredicateException
                    || exc.getCause() instanceof NoViableAltException) {
                final RecognitionException recognitionEx = (RecognitionException) exc.getCause();
                if (recognitionEx.getCtx() instanceof JavadocParser.HtmlTagContext) {
                    final Token htmlTagNameStart = getMissedHtmlTag(recognitionEx);
                    parseErrorMessage = new ParseErrorMessage(
                            errorListener.offset + htmlTagNameStart.getLine(),
                            MSG_JAVADOC_MISSED_HTML_CLOSE,
                            htmlTagNameStart.getCharPositionInLine(),
                            htmlTagNameStart.getText());
                }
            }

            if (parseErrorMessage == null) {
                // If syntax error occurs then message is printed by error listener
                // and parser throws this runtime exception to stop parsing.
                // Just stop processing current Javadoc comment.
                parseErrorMessage = errorListener.getErrorMessage();
            }

            result.setParseErrorMessage(parseErrorMessage);
        }

```

---

Also, it might be better to return an empty Javadoc tree in case of exceptions instead of rethrowing them, as a dummy code for now. This would help avoid build failures, such as those currently happening in JavadocPropertiesGenerator.